src/Button.js

```diff
@@ -1,11 +1,92 @@
-import React from 'react'
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import F from 'futil'
+import styled from '@emotion/styled'
+import { Subtitle, Text, Icon, Flex } from './'
+import theme from './theme'
 
-let Button = ({ isActive, primary, as: As = 'button', className, ...x }) => (
+let getColor = F.aliasIn(theme.colors)
+
+let buttonStyle = (baseColor, hoverColor, activeColor, textColor) => ({
+  backgroundColor: getColor(baseColor),
+  ':hover': { backgroundColor: getColor(hoverColor) },
+  ':active': { backgroundColor: getColor(activeColor) },
+  '& *': { color: getColor(textColor) },
+})
+
+let ButtonText = ({ compact = false, large = false, ...props }) =>
+  compact ? (
+    <Text extraSmall css={{ fontWeight: 'bold' }} {...props} />
+  ) : (
+    <Subtitle large={large} {...props} />
+  )
+
+let BaseButton = ({
+  as: As = 'button',
+  compact = false,
+  large = false,
+  icon,
+  children,
+  ...props
+}) => (
   <As
-    className={`gv-button ${isActive ? 'active' : ''} ${
-      primary ? 'primary' : ''
-    } ${className || ''}`}
-    {...x}
-  />
+    css={[
+      {
+        padding: large
+          ? theme.spaces[4]
+          : compact
+          ? `${theme.spaces[1]}px ${theme.spaces[2]}px`
+          : `${theme.spaces[2]}px ${theme.spaces[4]}px`,
+      },
+      icon && { paddingRight: 0 },
+    ]}
+    {...props}
+  >
+    <Flex alignItems="center">
+      <ButtonText {...{ compact, large }}>{children}</ButtonText>
+      {icon && (
+        <Icon
+          icon={icon}
+          style={{
+            paddingLeft: theme.spaces[1],
+            paddingRight: compact ? theme.spaces[1] : theme.spaces[2],
+            opacity: 0.5
+          }}
+        />
+      )}
+    </Flex>
+  </As>
 )
+
+let Button = styled(BaseButton)(
+  {
+    border: 'none',
+    outline: 'none',
+    cursor: 'pointer',
+    borderRadius: 3,
+    transition: 'background-color .2s linear',
+    ':active': { transition: 'none' },
+  },
+  buttonStyle('neutrals.4', 'neutrals.5', 'neutrals.6', 'secondaries.1'),
```

Comment:
This API isn’t great because you can’t tell at all from the invocation which color is which. In other scenarios, we’ve done object props, but in this case I think we should be defining the color pairings more semantically